lib/iris/fileformats/grib/__init__.py

```diff
@@ -958,12 +964,54 @@ def save_grib2(cube, target, append=False, **kwargs):
 
     # Save each latlon slice2D in the cube
     for slice2D in cube.slices([y_coords[0], x_coords[0]]):
-
-        # Save this slice to the grib file
         grib_message = gribapi.grib_new_from_samples("GRIB2")
         _save_rules.run(slice2D, grib_message)
-        gribapi.grib_write(grib_message, grib_file)
-        gribapi.grib_release(grib_message)
+        yield (slice2D, grib_message)
+
+
+def as_messages(cube):
+    """
+    Convert one or more cubes to GRIB messages.
+    Returns an iterable of grib_api GRIB messages.
+
+    Args:
+        * cube      - A :class:`iris.cube.Cube`, :class:`iris.cube.CubeList` or
+                      list of cubes.
+
+    """
+    return (message for cube, message in as_pairs(cube))
+
+
+def save_messages(messages, target, append=False):
+    """
+    Save a messages to a GRIB2 file.
+
+    Args:
+
+        * messages
```

Comment:
Needs a description. Perhaps, "An iterable of grib_api message IDs." Also needs a note (possibly in the function description) to explain that the messages will be released as part of the save.